### PR TITLE
[TM-35] Option To Apply For Pledge

### PR DESCRIPTION
**Pull request by mmiddleton3301**
Allowing the pulling of a single pledge.

See also: 

* https://github.com/SkillsFundingAgency/das-apim-endpoints/pull/414
* https://github.com/SkillsFundingAgency/das-levy-transfer-matching-web/pull/29
* https://github.com/SkillsFundingAgency/das-employer-config/pull/926
* https://github.com/SkillsFundingAgency/das-employer-config/pull/940
* https://github.com/SkillsFundingAgency/das-employer-config/pull/941

### CODE REVIEW

File: src/SFA.DAS.LevyTransferMatching.Api/Controllers/PledgesController.cs [hunk 1]
```diff
@@ -30,6 +31,29 @@ public async Task<IActionResult> GetPledges()
             return Ok(new GetPledgesResponse(result.Select(x => (GetPledgesResponse.Pledge)x)));
         }
 
+        [HttpGet]
+        [Route("pledges/{pledgeId}")]
+        [ProducesResponseType((int)HttpStatusCode.OK)]
+        [ProducesResponseType((int)HttpStatusCode.NotFound)]
+        public async Task<IActionResult> GetPledge(int pledgeId)
+        {
+            var result = await _mediator.Send(new GetPledgesQuery()
```
Comment (chrisfoster76) [inline src/SFA.DAS.LevyTransferMatching.Api/Controllers/PledgesController.cs:40]: The `GetPledge` controller should be sending a `GetPledgeQuery`, not repurposing another query for another use case.

**Review by chrisfoster76 — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.LevyTransferMatching.Api/Controllers/PledgesController.cs (hunk 1 above)
Comment (mmiddleton3301) [reply src/SFA.DAS.LevyTransferMatching.Api/Controllers/PledgesController.cs]: This has been changed.

File: src/SFA.DAS.LevyTransferMatching/Application/Queries/GetPledge/GetPledgeQueryHandler.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        }
+
+        public async Task<GetPledgeResult> Handle(GetPledgeQuery request, CancellationToken cancellationToken)
+        {
+            var pledges = await _dbContext.Pledges
+                .Where(x => x.Id == request.Id)
+                .Include(x => x.EmployerAccount)
+                .ToListAsync();
+
+            var pledge = pledges.SingleOrDefault();
```
Comment (chrisfoster76) [inline src/SFA.DAS.LevyTransferMatching/Application/Queries/GetPledge/GetPledgeQueryHandler.cs:28]: I think you could just to SingleOrDefaultAsync() above and not have to get a list to then get single.

Comment (chrisfoster76) [reply src/SFA.DAS.LevyTransferMatching/Application/Queries/GetPledge/GetPledgeQueryHandler.cs:28]: Oh also if it supports the cancellation token you can pass that in.

File: src/SFA.DAS.LevyTransferMatching/Application/Queries/GetPledge/GetPledgeResult.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿using SFA.DAS.LevyTransferMatching.Models;
+using System;
+
+namespace SFA.DAS.LevyTransferMatching.Application.Queries.GetPledge
+{
+    public class GetPledgeResult : Pledge
+    {
+        public object SingleOrDefault()
```
Comment (chrisfoster76) [inline src/SFA.DAS.LevyTransferMatching/Application/Queries/GetPledge/GetPledgeResult.cs:8]: I'm not sure this should be here.

**Review by chrisfoster76 — COMMENTED** [review]
Minor changes suggested

File: src/SFA.DAS.LevyTransferMatching/Application/Queries/GetPledge/GetPledgeResult.cs (hunk 3 above)
Comment (mmiddleton3301) [reply src/SFA.DAS.LevyTransferMatching/Application/Queries/GetPledge/GetPledgeResult.cs:8]: Nope, it shouldn't be - binned it.

File: src/SFA.DAS.LevyTransferMatching/Application/Queries/GetPledge/GetPledgeQueryHandler.cs (hunk 2 above)
Comment (mmiddleton3301) [reply src/SFA.DAS.LevyTransferMatching/Application/Queries/GetPledge/GetPledgeQueryHandler.cs]: Good spot. Changed.

**Review by chrisfoster76 — APPROVED** [review]

**Pull request merged**